deploy/tasks/pre-build-minimal.yaml

```diff
@@ -0,0 +1,72 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: pre-build-minimal
+  annotations:
+    tekton.dev/pipelines.minVersion: 0.12.1
+    tekton.dev/tags: image-build, konflux
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: docker
+spec:
+  description: |-
+    Sets up pre-build running the preprocessor, pushing the source and creating the OCI image.
+  params:
+    - name: IMAGE_URL
+      description: URL of the OCI image to create.
+      type: string
+    - name: NAME
+      description: Name of the pipeline run (i.e. unique dependency build name)
+      type: string
+    - name: BUILD_SCRIPT
+      description: The build script to embed with the Containerfile
+    - name: PREPROCESSOR_ARGS
+      description: The arguments for the build preprocessor
+    - name: ORAS_OPTIONS
+      type: string
+      description: Optional environment variable string for build-trusted-artifacts
+      default: ""
+  results:
+    - name: PRE_BUILD_IMAGE_DIGEST
+      description: Digest of the image just built
+    # - name: GIT_ARCHIVE
+    #   description: Git archive information
+  # stepTemplate:
+  #   computeResources: {}
+  #   volumeMounts:
+  #     - mountPath: /var/workdir/source
+  #       name: source
+  workspaces:
+    - name: source
+      description: The git repo will be cloned onto the volume backing this Workspace.
+      # mountPath: /var/workdir
+    # - name: tls
+  steps:
```

Comment:
Few comments:

- In https://github.com/redhat-appstudio/jvm-build-service/pull/2144 I've created a more 'minimal' version of the prebuild task (and also marking more parameters as optional). I think it would be better to try to adopt/improve that than creating another version?
- I'm not sure why you needed to remove the workspace mountpath?
- I agree tls workspace could be removed (potentially as another iteration).